script/generate_views

```diff
@@ -140,13 +140,23 @@ def create_views_if_not_exist(views, exclude, sql_dir):
 
         project, dataset, viewname = view.split(".")
         target = f"{view}_v{version}"
-        view_query = VIEW_QUERY_TEMPLATE.format(target=target).strip()
         view_dataset = dataset.rsplit("_", 1)[0]
         full_view_id = ".".join([project, view_dataset, viewname])
         target_file = os.path.join(sql_dir, view_dataset, viewname, "view.sql")
-        full_sql = f"CREATE OR REPLACE VIEW\n  `{full_view_id}`\nAS {view_query}\n"
 
         if not os.path.exists(target_file):
+            replacements = [
+                "`moz-fx-data-shared-prod.udf.normalize_metadata`(metadata) AS metadata",
+            ]
+            table = client.get_table(target)
+            if 'ping_info' in (f.name for f in table.schema):
+                # If we have ping_info, this must be a Glean schema.
```

Comment:
This inference of Glean schemas is that part of this that seems potentially dangerous. It seems unlikely that non-Glean pings would have a top-level `ping_info` field, but a miss here would cause the production schema pipeline to error out. I'm open to suggestions on better ways to determine "glean-ness", but this seems like a pragmatic choice for the time being.

Reply:
Maybe we could put something in the table's metadata?  We know when we generate the schema for the table whether it's glean or not, and maybe we could add something then?

Reply:
> I'm open to suggestions on better ways to determine "glean-ness", but this seems like a pragmatic choice for the time being.

glean datasets start with `org_mozilla_`, which doesn't require a `client.get_table` call to determine.

Reply:
> glean datasets start with org_mozilla_, which doesn't require a client.get_table call to determine.

I think the `org_mozilla_` prefix is an artifact of all Glean projects so far being Android projects. I don't know that we're guaranteed to have that prefix for all glean datasets in the future.

And in practice, it looks like the `client.get_table` call is returning quite quickly, so I think it's fine to introduce this API call here, as we already rely on the BQ API to list tables.

Reply:
> Maybe we could put something in the table's metadata? We know when we generate the schema for the table whether it's glean or not, and maybe we could add something then?

I agree this sounds like the right thing to do longer-term. I'll look into how feasible it would be to do so sooner rather than later.

Reply:
[I've put together a PR](https://github.com/mozilla-services/cloudops-infra/pull/1723) to add a `schema_id` label when deploying schemas. For Glean schemas, this would always be set to "moz://mozilla.org/schemas/glean/ping/1", and we would test for existence of such a label here.

Reply:
This change has been deployed to production so the label should be available, with the label value normalized [as discussed in that PR](https://github.com/mozilla-services/cloudops-infra/pull/1723#issuecomment-571730689]).

Reply:
We now use labels to determine this.